### PR TITLE
Made font, text boxes position, & logo position configurable Issue #154 & Issue  #155

### PR DESCRIPTION
**Pull request by abarber7**
Make font configurable  Issue #154
Make position of text boxes and logo configurable Issue #155

### CODE REVIEW

**Comment by McPringle** [comment]
Thank you very much for your two PRs, @abarber7, I highly appreciate your contributions!

You made some changes to the build file of the platform-dependent installer: ´install/gpx-animator.install4j´

I guess this is an error and is based on the refactoring tools of your IDE. These changes will break the installer builds.

**Comment by McPringle** [comment]
Why have you removed the 'photoAnimationDuration' configuration option?

**Comment by abarber7** [comment]
Hello, The changes to the build file of the platform-dependent installer: ´install/gpx-animator.install4j´ was an error.
Also, removing the 'photoAnimationDuration' configuration option was also an error.

**Comment by abarber7** [comment]
I have also separated the pull request into two different pull request for each issue.

**Comment by abarber7** [comment]
I have restored the install4j file, and photoAnimationDuration configuration option.

File: src/main/java/app/gpx_animator/Configuration.java
Comment (McPringle) [inline src/main/java/app/gpx_animator/Configuration.java:105]: Why did you change this constructor to be public? This class uses the [builder pattern](https://www.vogella.com/tutorials/DesignPatternBuilder/article.html), so nobody should call the constructor directly.

File: src/main/java/app/gpx_animator/Configuration.java
Comment (McPringle) [inline src/main/java/app/gpx_animator/Configuration.java:36]: If you really remove this constant, this class should have compile errors. This constant is used in two or three places in this class.

File: src/main/java/app/gpx_animator/Configuration.java
Comment (McPringle) [inline src/main/java/app/gpx_animator/Configuration.java:571]: If you move this method up above the `addTrackConfiguration` method, these changed lines will disappear and look like nothing changed here... 😊

File: src/main/java/app/gpx_animator/Option.java
Comment (McPringle) [inline src/main/java/app/gpx_animator/Option.java:49]: This line should not be removed

File: src/main/java/app/gpx_animator/Renderer.java
Comment (McPringle) [inline src/main/java/app/gpx_animator/Renderer.java:85]: If you create the enum I mentioned above, you can remove this.

File: src/main/java/app/gpx_animator/Renderer.java
Comment (McPringle) [inline src/main/java/app/gpx_animator/Renderer.java:251]: I renamed this class one or two days ago. After you pull from upstream, this change should be gone.

File: src/main/java/app/gpx_animator/Renderer.java
Comment (McPringle) [inline src/main/java/app/gpx_animator/Renderer.java:295]: I added this check and error message a couple of days ago. A pull from upstream should fix it, too.

File: src/main/java/app/gpx_animator/Renderer.java
Comment (McPringle) [inline src/main/java/app/gpx_animator/Renderer.java:312]: Reverting my fixed typo... 😁

File: src/main/java/app/gpx_animator/ui/GeneralSettingsPanel.java
Comment (McPringle) [inline src/main/java/app/gpx_animator/ui/GeneralSettingsPanel.java:78]: You should not remove this...

File: src/main/java/app/gpx_animator/ui/GeneralSettingsPanel.java
Comment (McPringle) [inline src/main/java/app/gpx_animator/ui/GeneralSettingsPanel.java:851]: Same here, please don't remove this line

File: src/main/java/app/gpx_animator/ui/GeneralSettingsPanel.java
Comment (McPringle) [inline src/main/java/app/gpx_animator/ui/GeneralSettingsPanel.java:922]: You know... 😊

**Review by McPringle — CHANGES_REQUESTED** [review]
I have left a lot of comments on your PR. If you need help solving these, please comment back! Most of the problems I have seen are a result of missing updates from the upstream. At the same time you worked on your issue, I was fixing a few others and we both changed the same files. With an update from upstream and a merge, most of this should fix itself.

**Comment by abarber7** [comment]
I attempted to merge the master into Antonio3 branch, and I am unsure if I did it correctly as I still manually resolved a few of the conflicts.

File: src/main/java/app/gpx_animator/Configuration.java
Comment (abarber7) [reply src/main/java/app/gpx_animator/Configuration.java:68]: I will look into this issue further, but I may need help with this one.

File: src/main/resources/i18n/Messages.properties
Comment (McPringle) [inline src/main/resources/i18n/Messages.properties:178]: This line should not be removed.

File: src/main/resources/i18n/Messages.properties
Comment (McPringle) [inline src/main/resources/i18n/Messages.properties:58]: This line should not be removed.

File: src/main/resources/i18n/Messages.properties
Comment (McPringle) [inline src/main/resources/i18n/Messages.properties:41]: This line should not be removed.

File: src/main/java/app/gpx_animator/ui/GeneralSettingsPanel.java
Comment (McPringle) [inline src/main/java/app/gpx_animator/ui/GeneralSettingsPanel.java:1117]: Don't add this line. This command is already there three lines above.

File: src/main/java/app/gpx_animator/ui/GeneralSettingsPanel.java
Comment (McPringle) [inline src/main/java/app/gpx_animator/ui/GeneralSettingsPanel.java:1016]: I moved this method to the `MapUtil` class three days ago. Just don't add it back to the `GeneralSettingsPanel` class. Remove it, and it will be fine.

File: src/main/java/app/gpx_animator/Renderer.java
Comment (McPringle) [inline src/main/java/app/gpx_animator/Renderer.java:651]: I moved this method to the `SpeedUtil` class. Just remove it here and it will be fine.

File: src/main/java/app/gpx_animator/Renderer.java
Comment (McPringle) [inline src/main/java/app/gpx_animator/Renderer.java:620]: I moved this method to the `SpeedUtil` class. Just remove it here and it will be fine.

**Review by McPringle — CHANGES_REQUESTED** [review]
Thank you very much for your continuous work on this PR, it looks much better now! There are still some issues, but very small ones. Your merge was nearly perfect. I spotted two or three methods in the `GeneralSettingsPanel` which should be removed. I moved these methods to different classes (`MapUtil` and `SpeedUtil`) and after your merge, they are back in the `GeneralSettingsPanel`. Just delete these methods, I added a comment there.

**Comment by abarber7** [comment]
Thank you for all of your help and hard work. I updated the GeneralSettingsPanel, Renderer, and Message.properties files.

**Comment by McPringle** [comment]
That looks much better! Great work! Now there should be only changes shown you made to solve this issue. I'll merge it after I had a few hours of sleep... :-)

File: src/main/java/app/gpx_animator/Configuration.java
Comment (McPringle) [reply src/main/java/app/gpx_animator/Configuration.java:68]: Sure! We can do this together if you like. With virtual pair programming.

File: src/main/java/app/gpx_animator/Configuration.java
Comment (McPringle) [inline src/main/java/app/gpx_animator/Configuration.java:101]: Just a simple formatting issue: Remove this blank line from the empty method body, please.

File: src/main/java/app/gpx_animator/Configuration.java
Comment (McPringle) [inline src/main/java/app/gpx_animator/Configuration.java:153]: Just a simple formatting issue: Remove this blank line, please.

File: src/main/java/app/gpx_animator/Configuration.java
Comment (McPringle) [inline src/main/java/app/gpx_animator/Configuration.java:296]: Just a simple formatting issue: The modified whitespace in front of the line makes it look like you changed four additional lines.

File: src/main/java/app/gpx_animator/Configuration.java
Comment (McPringle) [inline src/main/java/app/gpx_animator/Configuration.java:295]: You moved this line four lines up into the block of the position settings. I would prefer to have this line stay here together with the other photo related lines.

File: src/main/java/app/gpx_animator/Configuration.java
Comment (McPringle) [inline src/main/java/app/gpx_animator/Configuration.java:563]: Another formatting issue: Because of the modified whitespace it looks like you changed six lines of code.

File: src/main/java/app/gpx_animator/Renderer.java
Comment (McPringle) [inline src/main/java/app/gpx_animator/Renderer.java:214]: Looks like you removed the fix of bug #226. I guess this was a merge conflict that was resolved the wrong way.

File: src/main/java/app/gpx_animator/Renderer.java
Comment (McPringle) [inline src/main/java/app/gpx_animator/Renderer.java:448]: Looks like you removed the fix of bug #226. I guess this was a merge conflict that was resolved the wrong way.

File: src/main/java/app/gpx_animator/ui/GeneralSettingsPanel.java
Comment (McPringle) [inline src/main/java/app/gpx_animator/ui/GeneralSettingsPanel.java:950]: Just a simple formatting issue: Please remove this additional blank line.

File: src/main/java/app/gpx_animator/ui/GeneralSettingsPanel.java
Comment (McPringle) [inline src/main/java/app/gpx_animator/ui/GeneralSettingsPanel.java:957]: Just a simple formatting issue: Please correct this whitespace, because you did not modify the source code here.

**Review by McPringle — CHANGES_REQUESTED** [review]
I finished my review of this huge PR. You don't need to do the enum change, I can do it after the merge or, if you like, we can do it together via virtual pair programming (I can share my IDE).

I recommend next time you start working on an issue to write a comment on that issue. Then I know someone is working there and we can avoid changing the same files, so we don't have merge issues. Manual merges can sometimes be a pain in the ass - especially if you are not used to solving merge conflicts. :-)

If you solve the last few comments (skip my two comments about enum) I'll merge your PR to master.

**Review by McPringle — APPROVED** [review]
Hi @abarber7 -- I know it was hard work, thank you very much for your contribution, great work! I'll merge your PR now and will do a refactoring (using an enum for the positions) later.

**Comment by abarber7** [comment]
Thank you for all of your help. I am sorry that the PR was so huge, I tried to separate the issues into two different PR but, that was after I already made the huge PR. Sorry, for making you work so hard, but your efforts were greatly appreciated.
Next time I will comment to avoid the merge conflicts. Thank you again.

**Pull request merged**
